pkg/hintrunner/zero/zerohint_signatures.go

```diff
@@ -50,3 +53,81 @@ func createVerifyECDSASignatureHinter(resolver hintReferenceResolver) (hinter.Hi
 	}
 	return newVerifyECDSASignatureHinter(ecdsaPtr, signature_r, signature_s), nil
 }
+
+func newGetPointFromXHinter(xCube, v hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "VerifyECDSASignature",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> x_cube_int = pack(ids.x_cube, PRIME) % SECP_P
+			//> y_square_int = (x_cube_int + ids.BETA) % SECP_P
+			//> y = pow(y_square_int, (SECP_P + 1) // 4, SECP_P)
+			//> if ids.v % 2 == y % 2:
+			//>	 value = y
+			//> else:
+			//>	 value = (-y) % SECP_P
+			xCubeAddr, err := xCube.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			xCubeMemoryValues, err := hinter.GetConsecutiveValues(vm, xCubeAddr, 3)
+			if err != nil {
+				return err
+			}
+			v, err := hinter.ResolveAsFelt(vm, v)
+			if err != nil {
+				return err
+			}
+
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			secpBig, _ := secp_utils.GetSecPBig()
+
+			//> x_cube_int = pack(ids.x_cube, PRIME) % SECP_P
+			var xCubeValues [3]*fp.Element
+			for i := 0; i < 3; i++ {
+				xCubeValues[i], err = xCubeMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+			}
+			xCubeIntBig, err := secp_utils.SecPPacked(xCubeValues)
+			if err != nil {
+				return err
+			}
+			xCubeIntBig.Mod(xCubeIntBig, secpBig)
+
+			//> y_square_int = (x_cube_int + ids.BETA) % SECP_P
+			ySquareIntBig := new(big.Int).Add(xCubeIntBig, secp_utils.GetBetaBig())
+			ySquareIntBig.Mod(ySquareIntBig, secpBig)
+
+			//> y = pow(y_square_int, (SECP_P + 1) // 4, SECP_P)
+			y := new(big.Int).Exp(ySquareIntBig, new(big.Int).Div(new(big.Int).Add(secpBig, big.NewInt(1)), big.NewInt(4)), secpBig)
+			vBig := v.BigInt(new(big.Int))
```

Comment:
Personally I would split this up into multiple lines. It gets hard to read with nesting.

Reply:
moved to separate var called exponent